pkg/install/genevalogging.go

```diff
@@ -0,0 +1,399 @@
+	                    <RouteEvent eventName="CustomerAuditLogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	            </MdsdEvents>
+	        </Events>
+		</MonitoringManagement>`
+
+	t := template.Must(template.New("").Funcs(map[string]interface{}{
+		"Escape": func(s string) (string, error) {
+			var b bytes.Buffer
+			err := xml.EscapeText(&b, []byte(s))
+			return b.String(), err
+		},
+	}).Parse(tmpl))
+
+	b := &bytes.Buffer{}
+
+	err := t.Execute(b, map[string]string{
+		"Namespace":         genevaLoggingNamespace,
+		"Account":           genevaLoggingAccount,
+		"Region":            location,
+		"SubscriptionId":    subscriptionID,
+		"ResourceName":      clusterName,
+		"ResourceGroupName": resourceGroup,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return string(b.Bytes()), nil
+}
+
+func (i *Installer) installGenevaLogging(ctx context.Context) error {
+	g, err := i.loadGraph(ctx)
+	if err != nil {
+		return err
+	}
+
+	installConfig := g[reflect.TypeOf(&installconfig.InstallConfig{})].(*installconfig.InstallConfig)
+	resourceGroup := i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID[strings.LastIndexByte(i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')+1:]
+	mdsdConf, err := mdsdConfig(installConfig.Config.Azure.Region, resourceGroup, "subID", installConfig.Config.ClusterName, "GenevaLoggingNamespace", "GenevaLoggingAccount")
+	if err != nil {
+		return err
+	}
+
+	restConfig, err := restconfig.RestConfig(ctx, i.env, i.doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	cli, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	_, err = cli.CoreV1().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+		Spec: v1.NamespaceSpec{
+			Finalizers: []v1.FinalizerName{"kubernetes"},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	_, err = cli.CoreV1().ConfigMaps(namespace).Create(&v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fluentd-config",
+			Namespace: namespace,
+		},
+		Data: map[string]string{"fluent.conf": fluentConf},
+	})
+	if err != nil {
+		return err
+	}
+	_, err = cli.CoreV1().ConfigMaps(namespace).Create(&v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mdsd-config",
+			Namespace: namespace,
+		},
+		Data: map[string]string{"mdsd.xml": mdsdConf},
+	})
+	if err != nil {
+		return err
+	}
+	_, err = cli.CoreV1().Secrets(namespace).Create(&v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "azure-registry",
+			Namespace: namespace,
+		},
+		StringData: map[string]string{".dockerconfigjson": ""}, //todo
+	})
+	if err != nil {
+		return err
+	}
+	_, err = cli.CoreV1().Secrets(namespace).Create(&v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gcs-cert",
+			Namespace: namespace,
+		},
+		StringData: map[string]string{
+			"gcscert.pem": "",  //todo
+			"gcskey.pem":  ""}, //todo
```

Comment:
take these as parameters in rp-production.json, get them onto the RP disk and piped through to the RP probably via a mount+well known location rather than via environment variables.